### PR TITLE
[REF] extract portion that creates the custom field record

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code extraction

Before
----------------------------------------
Less readable

After
----------------------------------------
More readable

Technical Details
----------------------------------------
Note this incorporates https://github.com/civicrm/civicrm-core/pull/14705 & extends it (which in turn is a superset of https://github.com/civicrm/civicrm-core/pull/14724 - this allows a reviewer to choose the size of 'chunk' to review)

Comments
----------------------------------------
Towards https://github.com/civicrm/civicrm-core/pull/14694

### CODE REVIEW

**Comment by pradpnayak** [comment]
This need rebasing as well

**Comment by eileenmcnaughton** [comment]
thanks - this is now rebased

File: CRM/Core/BAO/CustomField.php [hunk 1]
```diff
@@ -152,42 +152,10 @@ public static function dataToHtml() {
...
-    $customField->copyValues($params);
-    $customField->save();
-
-    // make sure all values are present in the object for further processing
-    $customField->find(TRUE);
-
-    $triggerRebuild = CRM_Utils_Array::value('triggerRebuild', $params, TRUE);
-    //create/drop the index when we toggle the is_searchable flag
+    $customField = self::createCustomFieldRecord($params);
     $op = empty($params['id']) ? 'add' : 'modify';
```
Comment (eileenmcnaughton) [inline CRM/Core/BAO/CustomField.php]: Actually @pradpnayak the net effect here is to set the op AFTER the pre hook has been called. I'm now on the fence about that - I guess it's a good thing to set afterwards even though other BAO don't but are also kinda inconsistent.

**Comment by colemanw** [comment]
I don't have a strong opinion on that. I think this is better in than out so adding MOP.

**Comment by eileenmcnaughton** [comment]
merged per mop

**Pull request merged**